mypy/semanal.py

```diff
@@ -610,6 +610,13 @@ def _visit_overloaded_func_def(self, defn: OverloadedFuncDef) -> None:
             # redefinitions already.
             return
 
+        # Check final status, if at least one item or the implementation is marked
```

Comment:
I don't like this rule, as it allows a lot of flexibility with no clear benefit. I'd prefer to only allow `@final` in the implementation, and report an error if it's used elsewhere.